internal/latency/latency.go

```diff
@@ -67,6 +67,9 @@ func MatrixFrom(locations []string) Matrix {
 // Latency returns the latency between nodes a and b.
 // If a or b are not valid nodes, the function will panic.
 func (lm Matrix) Latency(a, b hotstuff.ID) time.Duration {
+	if a == b {
```

Comment:
Is this necessary? I think the latency matrix already has a 0 value for this entry?